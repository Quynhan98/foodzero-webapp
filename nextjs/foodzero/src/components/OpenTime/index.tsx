import { memo } from 'react'
import { Box, Flex, ListItem, Text, UnorderedList } from '@chakra-ui/react'

type BusinessHours = {
  meal: string
  time: string
}

interface OpenTimeProps {
  openTime: string
  businessHours: BusinessHours[]
}

const OpenTime = ({ openTime, businessHours }: OpenTimeProps) => {
  return (
    <Box minW="310px" maxW="696px">
      <Flex
        justifyContent="space-between"
        pb={{ base: '20px', '2xl': '38px' }}
        alignItems="center"
      >
        <Text
          variant="primary"
          fontSize={{ base: 'base', '2xl': 'md' }}
          fontWeight="bold"
          fontFamily="Rufina"
        >
          Open Time
        </Text>
        <Text variant="primary" size={{ base: 'small', '2xl': 'default' }}>
          {openTime}
        </Text>
      </Flex>
      <Box as="hr" color="black" borderTop="2px dashed white" />
      <UnorderedList
        display="flex"
        justifyContent="space-between"
        gap={{ base: '20px', '2xl': '133px' }}
        pt={{ base: '20px', '2xl': '38px' }}
        listStyleType="none"
        marginLeft="0px"
      >
        {businessHours.map((item) => (
          <ListItem key={`business-hours-${item.meal}`}>
            <Text variant="primary" size={{ base: 'small', '2xl': 'default' }}>
              {item.meal}
            </Text>
            <Text variant="primary" size={{ base: 'small', '2xl': 'default' }}>
              {item.time}
            </Text>
          </ListItem>
        ))}
      </UnorderedList>
    </Box>
  )
}

export default memo(OpenTime)
