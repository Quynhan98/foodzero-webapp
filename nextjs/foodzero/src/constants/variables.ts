export const NAV_LIST = [
  {
    value: 1,
    name: 'Home',
    href: '/',
  },
  {
    value: 2,
    name: 'Menu',
    href: '/',
  },
  {
    value: 3,
    name: 'About',
    href: '/',
  },
  {
    value: 4,
    name: 'Contact',
    href: '/',
  },
]

export const SOCIAL_LIST = [
  {
    value: 1,
    src: '/images/instagram.webp',
    alt: 'Instagram',
    link: '/',
  },
  {
    value: 2,
    src: '/images/twitter.webp',
    alt: 'Twitter',
    link: '/',
  },
  {
    value: 3,
    src: '/images/facebook.webp',
    alt: 'Facebook',
    link: '/',
  },
  {
    value: 4,
    src: '/images/youtube.webp',
    alt: 'youtube',
    link: '/',
  },
]

export const CONTACT = {
  phoneNumber: '+86 852 346 000',
  email: 'info@foodzero.com',
  address: '1959 Sepulveda Blvd. Culver City, CA, 90230',
}

export const OPEN_TIME = {
  openTime: 'Sunday - Friday',
  businessHours: [
    { meal: 'Brunch', time: '11:00–12:00' },
    { meal: 'Lunch', time: '13:00–17:00' },
    { meal: 'Dinner', time: '18:00–20:00' },
  ],
}

export enum BREAKPOINTS {
  SMALL = '(max-width: 320px)',
  MEDIUM = '(max-width: 768px)',
  LARGE = '(max-width: 960px)',
  EXTRA_LARGE = '(max-width: 1200px)',
}

export const RESERVATION_TIME = [
  '11:00 am',
  '12:00 am',
  '13:00 pm',
  '14:00 pm',
  '15:00 pm',
  '16:00 pm',
  '17:00 pm',
  '18:00 pm',
  '19:00 pm',
]

export const NUMBER_OF_PERSON = [
  '1 Person',
  '2 Person',
  '3 Person',
  '4 Person',
  '5 Person',
  '6 Person',
  '7 Person',
  '8 Person',
  '9 Person',
  '10 Person',
]

export const LOCAL_STORAGE_KEY = {
  USER_ID: 'userId',
  IS_TOKEN: 'isToken',
}
