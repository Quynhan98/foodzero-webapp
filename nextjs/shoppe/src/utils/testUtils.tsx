import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
