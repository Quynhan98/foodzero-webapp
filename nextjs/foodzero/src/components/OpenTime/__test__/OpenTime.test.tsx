import { render, screen } from '@utils/testUtils'

// Components
import OpenTime from '@components/OpenTime'

// Constants
import { OPEN_TIME } from '@constants/variables'

describe('OpenTime component', () => {
  const props = { ...OPEN_TIME }

  it('Renders OpenTime default correctly', () => {
    const { container } = render(<OpenTime {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should render data correctly', () => {
    render(<OpenTime {...props} />)

    const openTime = screen.getByText(props.openTime)
    const brunch = screen.getByText(props.brunch)
    const lunch = screen.getByText(props.lunch)
    const dinner = screen.getByText(props.dinner)

    expect(openTime).toHaveTextContent(props.openTime)
    expect(brunch).toHaveTextContent(props.brunch)
    expect(lunch).toHaveTextContent(props.lunch)
    expect(dinner).toHaveTextContent(props.dinner)
  })
})
