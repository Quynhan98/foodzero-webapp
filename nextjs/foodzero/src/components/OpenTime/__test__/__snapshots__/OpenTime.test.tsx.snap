// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenTime component Renders OpenTime default correctly 1`] = `
<div>
  <div
    class="css-1ygbazq"
  >
    <div
      class="css-34hfx4"
    >
      <p
        class="chakra-text css-sqdrsl"
      >
        Open Time
      </p>
      <p
        class="chakra-text css-1xhkjsw"
      >
        Sunday - Friday
      </p>
    </div>
    <hr
      class="css-1iijbtl"
    />
    <ul
      class="css-1o7naxv"
      role="list"
    >
      <li
        class="css-0"
      >
        <p
          class="chakra-text css-1xhkjsw"
        >
          Brunch
        </p>
        <p
          class="chakra-text css-1xhkjsw"
        >
          11:00–12:00
        </p>
      </li>
      <li
        class="css-0"
      >
        <p
          class="chakra-text css-1xhkjsw"
        >
          Lunch
        </p>
        <p
          class="chakra-text css-1xhkjsw"
        >
          13:00–17:00
        </p>
      </li>
      <li
        class="css-0"
      >
        <p
          class="chakra-text css-1xhkjsw"
        >
          Dinner
        </p>
        <p
          class="chakra-text css-1xhkjsw"
        >
          18:00–20:00
        </p>
      </li>
    </ul>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
